rapid/workflow/queue_handlers/container_handlers/container_handler.py

```diff
@@ -0,0 +1,30 @@
+from abc import ABCMeta, abstractproperty
+
+from rapid.lib.work_request import WorkRequest
+from rapid.workflow.queue_handlers.queue_handler import QueueHandler
+
+
+class ContainerHandler(QueueHandler):
+    __metaclass__ = ABCMeta
+
+    @abstractproperty
+    def container_identifier(self):
+        yield
+
+    def can_process_work_request(self, work_request):
+        # type: (WorkRequest) -> bool
+        try:
+            (grain_type, image) = self._get_grain_type_split(work_request.grain)
+            return grain_type == self.container_identifier
+        except ValueError:
+            pass
+        return False
```

Comment:
detecting whether this work request is handled by a container. Should match <container_type>://<image>. Question: what happens if grain is "ecs" or "docker"? Will the queue say it can handle it, then fail? Should it _require_ that image not be empty? Or will python treat that as an error?

Reply:
Great thoughts! The image will be used later. If that process doesn't work, calling to ECS, then the system will detect it and mark the work_request failed. That scenario is accounted for. I could make the logic not call ECS unless there is an image. I will do that.

Reply:
This has been accounted for in another push of code.